samples/csharp/web/avatar/Services/SpeechTokenService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Avatar.Models;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Options;
+
+namespace Avatar.Services
+{
+    public class SpeechTokenService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ClientSettings _clientSettings;
+
+        public SpeechTokenService(HttpClient httpClient, IOptions<ClientSettings> clientSettings)
+        {
+            _httpClient = httpClient;
+            _clientSettings = clientSettings.Value;
+        }
+
+        public async Task RefreshSpeechTokenAsync(CancellationToken stoppingToken)
+        {
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                if (!string.IsNullOrEmpty(_clientSettings.SpeechPrivateEndpoint))
+                {
+                    var credential = new DefaultAzureCredential(new DefaultAzureCredentialOptions
+                    {
+                        ManagedIdentityClientId = _clientSettings.UserAssignedManagedIdentityClientId
+                    });
+
+                    var token = await credential.GetTokenAsync(new TokenRequestContext(new[] { "https://cognitiveservices.azure.com/.default" }));
+                    GlobalVariables.SpeechToken = $"aad#{_clientSettings.SpeechResourceUrl}#{token.Token}";
+                    Console.WriteLine("Token refreshed using managed identity.");
+                }
+                else
+                {
+                    var url = $"https://{_clientSettings.SpeechRegion}.api.cognitive.microsoft.com/sts/v1.0/issueToken";
+                    var request = new HttpRequestMessage(HttpMethod.Post, url);
+                    request.Headers.Add("Ocp-Apim-Subscription-Key", _clientSettings.SpeechKey);
+
+                    var response = await _httpClient.SendAsync(request);
+                    response.EnsureSuccessStatusCode();
+
+                    GlobalVariables.SpeechToken = await response.Content.ReadAsStringAsync();
+                    Console.WriteLine("Token refreshed using subscription key.");
```

Comment:
Please rename subscription key to API key

Reply:
updated